fudge-ratio/src/main/webapp/script.js

```diff
@@ -66,22 +68,62 @@ buttonClear.addEventListener("click", () => {
   if (button.textContent === "Stop") button.click();
   (seconds = 0), (minutes = 0), (hours = 0);
   updateText();
+  cleanSessionState();
 });
 
 const formatTaskString = () => {
-    let dropdown = document.getElementById("dropdown-contents");
-    let op = dropdown.options[dropdown.selectedIndex];
-	let category = op.parentNode.label;
-    let formattedString = "";
-    if (category == "Schoolwork"){
-        formattedString = task + " homework!";
-    }
-    else if (category == "Job-related tasks"){
-        formattedString = "doing " + task + "!" ;
-    }
-    else formattedString = task + "!"; 
-    return formattedString; 
-}
+  let op = dropdown.options[dropdown.selectedIndex];
+  let category = op.parentNode.label;
+  let formattedString = "";
+  if (category == "Schoolwork") {
+    formattedString = task + " homework!";
+  } else if (category == "Job-related tasks") {
+    formattedString = "doing " + task + "!";
+  } else formattedString = task + "!";
+  return formattedString;
+};
 const displayMessage = () => {
-    document.getElementById("finalMessage").innerHTML = "Next time please allocate" + duration + " for "  + formatTaskString();
+  document.getElementById("finalMessage").innerHTML =
+    "Next time please allocate" + duration + " for " + formatTaskString();
+};
+
+// Save state in local Store ====
+const timeKey = "startingTime";
+const indexKey = "selectedIndex";
+
+const saveSessionState = () => {
+  if (informationThere(recoverSessionState())) return;
+  localStorage.setItem(timeKey, Date.now());
+  localStorage.setItem(indexKey, dropdown.selectedIndex);
+};
+
+const recoverSessionState = () => {
+  return [localStorage.getItem(timeKey), localStorage.getItem(indexKey)];
 };
+
+const cleanSessionState = () => {
+  localStorage.removeItem(timeKey);
+  localStorage.removeItem(indexKey);
+};
+
+const informationThere = array => array.every(x => x != null);
+
+if (informationThere(recoverSessionState())) {
+  const [timeStarted, selectedIndex] = recoverSessionState();
+  const start = parseInt(timeStarted);
+  dropdown.selectedIndex = selectedIndex;
+
+  const now = Date.now();
+  const elapsed = Math.floor((now - start) / 1000);
+
+  seconds = elapsed % 60;
+  minutes = Math.floor(elapsed / 60) % 60;
+  hours = Math.floor(elapsed / 3600) % 24;
+  updateText();
+  button.click();
+}
+
+dropdown.addEventListener("change", () => {
+  const current = recoverSessionState()[1];
+  if (current) dropdown.selectedIndex = current
+});
```

Comment:
I was going to suggest that we do this so that the user cant change the dropdown while in the middle of timing. Thank you for doing this!